### PR TITLE
Add gloo-console-timer

### LINKED ISSUE #16: RAII type for console.time and console.timeEnd
Let's pull it out of https://rustwasm.github.io/book/game-of-life/time-profiling.html#time-each-universetick-with-consoletime-and-consoletimeend

### PR DESCRIPTION
**Pull request by cbrevik**
Resolves #16 

Adds the `gloo-console-timer` crate with the `ConsoleTimer` RAII type.

Docs feature the preferred `scope`-method for measurement:
```rust
let value = ConsoleTimer::scope("foo", || {
    // Place code to be measured here
});
```

Was a bit unsure how to write tests here, since it seems I cannot verify through tests that the measurement including label is printed to the console.

### CODE REVIEW

File: crates/console-timer/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,75 @@
+/*!
+
+Measuring invocation time on the Web can be done with `console.time`
+and `console.timeEnd`.
```
Comment (fitzgen) [inline crates/console-timer/src/lib.rs]: Nitpick: can we link to MDN for more context here? https://developer.mozilla.org/en-US/docs/Web/API/console#Timers

And also mention that the timings end up in the browser's developer tools console and sometimes their performance view? I think we can crib a lot of this text from the rustwasm book: https://rustwasm.github.io/docs/book/reference/time-profiling.html#the-consoletime-and-consoletimeend-functions

File: crates/console-timer/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+This API wraps both the `time` and `timeEnd` calls into a single type
+named `ConsoleTimer`, ensuring both are called.
+
+## Scoped measurement
```
Comment (fitzgen) [inline crates/console-timer/src/lib.rs]: Can we also have a section on the RAII-style usage below this? Potentially showing timing asynchronous operations, where the scoped callbacks approach can't be used.

File: crates/console-timer/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+[package]
+name = "gloo-console-timer"
+version = "0.1.0"
+authors = ["Christian Brevik <christian.brevik@gmail.com>"]
```
Comment (fitzgen) [inline crates/console-timer/Cargo.toml]: Can you make this `"The Rust and WebAssembly Working Group"` like the root `Cargo.toml` does?

I just realized that the `crates/timers/Cargo.toml` doesn't use that same author line, and I'll file a PR to fix it.

Comment (cbrevik) [reply crates/console-timer/Cargo.toml]: Woops, didn't check the root `Cargo.toml`. Will fix!

**Comment by cbrevik** [comment]
Adressed feedback, but I see one of the doc tests doesn't build, so will have to fix that first. (The one using `gloo_timers`).

**Comment by cbrevik** [comment]
Maybe there is a better way to handle this than to `ignore` the test? 

I didn't want to add dependencies just to make that example compile.

File: crates/console-timer/src/lib.rs [hunk 4]
```diff
@@ -32,7 +32,7 @@ For scenarios where `ConsoleTimer::scope` can't be used, like with
 asynchronous operations, you can use `ConsoleTimer::new` to create a timer.
 The measurement ends when the timer object goes out of scope / is dropped.
 
-```no_run
+```ignore
```
Comment (fitzgen) [inline crates/console-timer/src/lib.rs]: What about adding a `dev-dependency`? Exists for precisely this kind of thing ;)

Comment (cbrevik) [reply crates/console-timer/src/lib.rs]: Hehe alright, not clear on what best practices are in cases like these. Done! 👍

**Pull request merged**